ospfd/ospf_abr.c

```diff
@@ -1788,6 +1788,28 @@ void ospf_abr_task(struct ospf *ospf)
 		zlog_debug("ospf_abr_task(): Stop");
 }
 
+/* If there's redistribution configured, we need to refresh external
+ * LSAs in order to install Type-7 and flood to all NSSA Areas
+ */
+static void ospf_abr_nssa_redist_task(struct ospf *ospf)
+{
+	int type;
+
+	for (type = 0; type < ZEBRA_ROUTE_MAX; type++) {
+		struct list *red_list;
+		struct listnode *node;
+		struct ospf_redist *red;
+
+		red_list = ospf->redist[type];
+		if (!red_list)
+			continue;
+
```

Comment:
I think , default information also has to be handled when NSSA enabled. Because, earlier default route would have originated as type-5 if default-information origination configured now it is has to originate as type-7 since it is NSSA.

Reply:
Added